### PR TITLE
Feature - Member Access Controls

### PR DESCRIPTION
**Pull request by chetansatasiya**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Theme PR
https://github.com/buddyboss/buddyboss-theme/pull/1029

### Platform Pro PR
https://github.com/buddyboss/buddyboss-platform-pro/pull/84

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/buddypress/document/index.php [hunk 1]
```diff
@@ -40,7 +40,7 @@
 		<?php
 		bp_get_template_part( 'common/search-and-filters-bar' );
 
-		if ( ! empty( $active_extensions ) && bp_is_profile_document_support_enabled() && is_user_logged_in() ) {
+		if ( ! empty( $active_extensions ) && bp_is_profile_document_support_enabled() && is_user_logged_in() && apply_filters( 'bp_user_can_create_document', true ) ) {
```
Comment (dipskakadiya) [inline src/bp-templates/bp-nouveau/buddypress/document/index.php:43]: @chetansatasiya  Not mandatory but just check if we can do that.

`apply_filters( 'bp_user_can_create_document', true )` if we using same filter multiple place can't we create function for that and use it. Like BuddyPress doing that.

**Review by dipskakadiya — DISMISSED** [review]

**Review by dipskakadiya — DISMISSED** [review]
@chetansatasiya Approved. Please merge with `dev` and assign for QA

**Comment by sumitbuddyboss** [comment]
Hello @chetansatasiya 

Group type - Profile Type settings are working as expected, For proof please check the below video.
https://www.loom.com/share/72883e60b87c42c58f9707d25d080f0c


cc @mohit-buddyboss

**Review by mehulkaklotar — APPROVED** [review]
looks good! please resolve conflicts and merge to dev

**Pull request merged**